src/Nethermind/Nethermind.Optimism/OptimismPayloadPreparationService.cs

```diff
@@ -50,6 +54,20 @@ protected override void ImproveBlock(string payloadId, BlockHeader parentHeader,
             _payloadStorage.TryAdd(payloadId,
                 new NoBlockImprovementContext(currentBestBlock, UInt256.Zero, startDateTime));
         }
-        else base.ImproveBlock(payloadId, parentHeader, payloadAttributes, currentBestBlock, startDateTime);
+        else
+        {
+            if (payloadAttributes is OptimismPayloadAttributes optimismPayload)
+            {
+                var spec = _specProvider.GetSpec(currentBestBlock.Header);
+                if (spec.IsOpHoloceneEnabled)
+                {
+                    EIP1559Parameters eip1559Parameters = optimismPayload.DecodeEIP1559Parameters();
```

Comment:
This could fail and throw exception

Reply:
I think that at that point we've already validated the attributes so there should not be a risk of exceptions (unless I'm missing something):

1. We validate attributes in
https://github.com/NethermindEth/nethermind/blob/4013c7ab406c13b34c03740e46dcd438461cccda/src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs#L89-L95

2. Then we use the `IPreparationService` in `StartBuildingPayload`
https://github.com/NethermindEth/nethermind/blob/4013c7ab406c13b34c03740e46dcd438461cccda/src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs#L327

3. Eventually, `OptimismPayloadPreparationService.ImproveBlock` is called:
https://github.com/NethermindEth/nethermind/blob/4013c7ab406c13b34c03740e46dcd438461cccda/src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PayloadPreparationService.cs#L74-L85

https://github.com/NethermindEth/nethermind/blob/4013c7ab406c13b34c03740e46dcd438461cccda/src/Nethermind/Nethermind.Optimism/OptimismPayloadPreparationService.cs#L45-L46